### ISSUE TITLE
Keep getting "Unable to open <file>: File not found" when clicking on Problem links

### ISSUE THREAD

**Issue by StephenWeatherford** [microsoft/vscode#50426]
Issue Type: <b>Bug</b>

I'm not currently sure of an exact repro.  If this issue is not known, I can try to come up with an exact repro.  But this is happening quite a lot with at least my current project, but didn't used to happen before the recent update.

What's happening is that I'm getting a lot of problems that don't go away even when fixed, and clicking on it gives the error "Unable to open …: File not found", and asks if I want to create the file.  But the file does in fact exist.

The only way I know of to get rid of them is restarting vscode.  

Still repros with all extensions disabled.

VS Code version: Code 1.23.1 (d0182c3417d225529c6d5ad24b7572815d0de9ac, 2018-05-10T17:11:17.614Z)
OS version: Windows_NT x64 10.0.17134

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Xeon(R) CPU E31230 @ 3.20GHz (4 x 3193)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>rasterization: disabled_software<br>video_decode: enabled<br>video_encode: enabled<br>vpx_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Memory (System)|15.96GB (5.48GB free)|
|Process Argv|C:\Program Files\Microsoft VS Code\Code.exe C:\Repos\vscode-azurestorage|
|Screen Reader|no|
|VM|0%|

</details><details><summary>Extensions (22)</summary>

Extension|Author (truncated)|Version
---|---|---
mongogo|bag|0.1.2
regex|chr|0.2.0
vscode-markdownlint|Dav|0.17.0
githistory|don|0.4.1
jupyter|don|1.1.4
tslint|eg2|1.0.29
vscode-bump|fab|1.5.1
docthis|joe|0.6.0
vscode-antlr4|mik|2.0.3
vscode-azureappservice|ms-|0.8.1
vscode-azurefunctions|ms-|0.9.1
vscode-azurestorage|ms-|0.3.1
vscode-cosmosdb|ms-|0.7.1
python|ms-|2018.4.0
azure-account|ms-|0.4.0
csharp|ms-|1.15.2
azurerm-vscode-tools|msa|0.4.2
vscode-versionlens|pfl|0.21.0
yo|sam|0.9.3
extension-api|spy|0.3.0
python|tht|0.2.3
code-lens-roundup|wad|1.0.0

</details>

**Comment by StephenWeatherford** [comment]
![image](https://user-images.githubusercontent.com/6913354/40520393-8ed74d24-5f7a-11e8-9c2e-d082a38308e5.png)

**Comment by mjbvz** [comment]
Are these errors coming from the editor or from a problem matcher in a build task?

**Comment by StephenWeatherford** [comment]
@mjbvz How do I tell?  I'm guessing problem matcher, it's a standard NodeJS project (a VS extension), here's my tasks.json:

```
{
    "version": "2.0.0",
    "tasks": [
        {
            "type": "npm",
            "script": "watch",
            "problemMatcher": "$tsc-watch",
            "isBackground": true,
            "presentation": {
                "reveal": "never"
            },
            "group": {
                "kind": "build",
                "isDefault": true
            }
        },
        {
            "type": "npm",
            "script": "lint",
            "problemMatcher": {
                "base": "$tslint5",
                "fileLocation": "absolute"
            }
        }
    ]
}

```

**Comment by mjbvz** [comment]
When do the errors show up? When you open a file or when you start the build task? 

Also, do you see this in vs code insiders?

**Comment by StephenWeatherford** [comment]
I think a problem matcher.  The links work fine at first, it's only after I've done some coding (maybe debugging) that they stop linking correctly and don't go away.

Repros in Insiders as well.

**Comment by dbaeumer** [comment]
@StephenWeatherford best would be to provide a GitHub repository we can clone that demos this. I looked at the screen shot and the file path actually does look fine to me.

**Comment by dbaeumer** [comment]
@cpadilla in contrast to @StephenWeatherford examples your pathes are actually broken. How to you compile the project ?

And can you provide the output when the task is run. The output seems to have a `1>` character in it which gets added to the file path which makes the file not being correct.

**Comment by cpadilla** [comment]
I'm using devenv to compile my project. It usually outputs `1>` before each line so I don't know how I would change this. Isn't there a way I can change how VSCode scans the output for errors?

![code_2018-10-05_13-25-35](https://user-images.githubusercontent.com/1704172/46550212-5d472180-c8a2-11e8-84ba-cba88e9dec42.png)

**Comment by dbaeumer** [comment]
Which problem matcher have you attached to the build task. The standard $mscompile matcher actually hanldes this. It has the following regex
```js
/^(?:\s+\d+\>)?([^\s].*)\((\d+|\d+,\d+|\d+,\d+,\d+,\d+)\)\s*:\s+(error|warning|info)\s+(\w{1,2}\d+)\s*:\s*(.*)$/
```

**Comment by hackzfy** [comment]
![image](https://user-images.githubusercontent.com/14977369/47345622-ca014f00-d6dd-11e8-96d3-6de23a500d0b.png)

**Comment by StephenWeatherford** [comment]
I still get this sporadically, but don't know how to repro consistently. It does however seem to happen most often after deleting a file or making refactoring types of changes.

**Comment by Tyriar** [comment]
@hackzfy that form works for me:

<img width="470" alt="screen shot 2018-10-24 at 7 50 17 am" src="https://user-images.githubusercontent.com/2193314/47439590-7df7ee00-d761-11e8-80d8-dd8ff3466483.png">

Please update to latest and create a new issue if you still see this as it's unrelated to the problems feature.

**Comment by dbaeumer** [comment]
Reopening since @StephenWeatherford problem is not solved.

**Comment by amerzafar** [comment]
I have same problem in running simple .net core api on mac

**Comment by gpassero** [comment]
I'm having the issue here as well. I noticed that in my case VSCode is trying to open a file with the wrong name. e.g.
```
Unable to open 'File.ts[6,': File not found (file:///c:/project/File.ts[6,).
```

The tslint error that I'm clicking on:

```
ERROR: C:/project/File.ts[6, 7]: 'foo' is declared but its value is never read.
```

**Comment by dbaeumer** [comment]
@gpassero is this in the terminal or the problems view?

**Comment by jheffez** [comment]
I'm having the same problem. In my case it's with python. 

On a side note - I noticed that this thread has been opened over 6 months ago and yet the still trying to figure out what is going on...

**Comment by dbaeumer** [comment]
@StephenWeatherford problem (who opened the original problem) is still not reproducible for us (see https://github.com/Microsoft/vscode/issues/50426#issuecomment-432384304). 

@jheffez since your problem is with Python it is very likely unrelated to this issue. In all cases steps to reproduce a problem are always very helpful to understand what the underlying issue is.

**Comment by jheffez** [comment]
@StephenWeatherford How did you decide that this is python problem? if you read carefully my original post you’ll notice that I specifically wrote that if I run the code without stepping in/out (F10, F11, etc) it works. It’s only when I take my time to debug that it pops up.

In addition after doing some googling it's clear that this issue has been in existence since last year and not just in Python development. Therefore, it is not isolated to a specific lang/environment. 

It’s hard to reproduce (just tried a few times). I will update the post once I get more on that. In the meantime it would be great to get feedback from others on the matter.

**Comment by StephenWeatherford** [comment]
@jheffez That comment was from @dbaeumer, not me.  I see this only very rarely now, so I assume my scenario was fixed or improved.  If you can provide them a way to reproduce or at least increase the chances of it happening, that would help them.

**Comment by dbaeumer** [comment]
@jheffez I am a little puzzled here. The only post I know from you is this one: https://github.com/Microsoft/vscode/issues/50426#issuecomment-448133617 which claims you are seeing this in Python.

And it depends on how a problem got created to understand why the link is not working. If it came from an extension we need to find the extension producing the problem. If it came from a task run it is typically the problem matcher. If it is in the terminal it might be the link detection in the terminal.

This is why steps are always helpful.

@StephenWeatherford steps to reproduce what you are seeing are still highly appreciated.

**Comment by gpassero** [comment]
@dbaeumer  The problem I mentioned occurs specifically if I try to commit changes using the VSCode UI, for links related to both lint (with tslint) and compilation (with tsc) errors. The output is displayed in Output> Git. The problem doesn't occur if I run `tslint`, `tsc`  or `git commit` (which calls the pre-commit hook with tslint and tsc) in the terminal.

**Comment by StephenWeatherford** [comment]
If I see it again, I'll try, but it's never been obvious how to get it to reproduce.

**Comment by dbaeumer** [comment]
@gpassero I am a little puzzled. Were do you click the link ?

@StephenWeatherford do you some git branch switching before that happens?

**Comment by gpassero** [comment]
@dbaeumer I go to the version control side bar, stage my changes then click the check sign to commit it. The problem happens then.

**Comment by gpassero** [comment]
My problem is a bit different from the reported in this issue. I'll search other issues for my problem and maybe create a new one if I can't find an exact match.

**Comment by Vaporwave30** [comment]
I'm late to the party, but the same error happens to me too. I have a file named "input.txt" in the same directory the Python script is in.
```python
open("input.txt", 'r').read().splitlines()
```
gives me
> `FileNotFoundError: [Errno 2] No such file or directory: 'input.txt'`

screenshot:
![image](https://user-images.githubusercontent.com/43111111/53409887-df0f2700-39ca-11e9-9028-fe6ef43ea3f5.png)

**Comment by unmultimedio** [comment]
Same here! Actually a piece of info that may be useful, only happens when some place in the path there's a dash, let's say my file is `/Users/unmultimedio/go/src/github.com/foo-bar/baz.go` and it keeps trying to parse it as `(file:///Users/unmultimedio/go/src/github.com/foo/-bar/baz.go)`

**Comment by dbaeumer** [comment]
@unmultimedio are you using a test runner ?

**Comment by Aypahyo** [comment]
I ran into the same issue.
I am using an E2C instance as a filesystem through the SSH FS plugin.
All files are on the topmost level without directories.
For a build I ssh into the machine and execute a shell script.

```
ls;
docker run --rm -v "$PWD":/usr/src/myapp -w /usr/src/myapp gcc make
```

the output is 

> build.sh
> main
> main.cpp
> makefile
> g++ -Wall -std=c++17 -O3 -o main main.cpp
> main.cpp: In function 'int main()':
> main.cpp:5:16: error: unable to find numeric literal operator 'operator""f'
>      return int{0f};
>                 ^~
> main.cpp:5:16: note: use -fext-numeric-literals to enable more built-in suffixes
> makefile:2: recipe for target 'main.o' failed
> make: *** [main.o] Error 1
> The terminal process terminated with exit code: 1

when I open the offending file myself the OPEN EDITORS reads: 

> main.cpp \

when I inspect the file in the PROBLEMS tab a message pops up reading 

> Unable to open 'main.cpp': File not found (file:///main.cpp).

My E2C machine only has port 22 for ssh open.

so in principle the URI reads correct

When I right click on main.cpp to copy the Path I get 

` \main.cpp`

I assume that the mechanism to open files is not based on the workspace.

**Comment by motor75** [comment]
I was about to enter this as a bug when I found this thread; I've been having this issue since I started using VS Code a year ago.  My local GIT repo has dashes in the path so maybe this is a problem with VS Code whereby it can't resolve folder names with dashes...see below.  Is there any update to this or a fix forthcoming?  I'm sure lots of people have dashes in their path names and are experiencing this issue.

Here's the path for me...
C:\Users\moserk\Documents\GitHub\CDM .NET Core\eti-cdm\messagelog\messagelog.API\GetMessageLog.cs

Here's a screen shot of clicking on an error in the output pane and getting a message that the file to which the error pertains to does not exist (when it most certainly does):
![image](https://user-images.githubusercontent.com/42443455/56322357-dae3c800-6136-11e9-9c2e-d82974e79d3c.png)

**Comment by unmultimedio** [comment]
@dbaeumer Yes, I'm running the `go test` tool that's included in the go `*_test.go` files:

![__go_src_code_hq_twilio_com_authy_trusted-comms-api_controller_reachers_test_go_—_authy-service__Workspace_](https://user-images.githubusercontent.com/4431392/56325418-bd632e00-6137-11e9-9c75-5b6fdc936fa5.png)

```
Running tool: /usr/local/bin/go test -timeout 30s path-to-the-folder/where-my-go-files-are -coverprofile=/var/folders/some-very-long-cover-folder/go-code-cover
```

**Comment by dbaeumer** [comment]
I tested this using TS and VS Code in general has no problems with `-` in paths or names. 

![capture](https://user-images.githubusercontent.com/1931590/56351344-319be100-61cd-11e9-95cc-1d2b7cb02364.png)

@unmultimedio / @motor75 can you provide me with a GitHub repository I can clone that demos this. Then I can see if VS Code has a problem with the setup you are using or if this might be related to a extension.

**Comment by motor75** [comment]
@dbaeumer This seems to have been resolved with an issue that I posted regarding OmniSharp which seems to have been caused by a corrupt or missing installation of .NET 4.7.2.  See this [issue ](https://github.com/OmniSharp/omnisharp-vscode/issues/3004) in the OmniSharp repository for details.  Since this has been fixed, I haven't had any further problems with opening files so far, but if I do I'll post my results here.

**Comment by maxim-ge** [comment]
Seems VS Code adds a wrong folder segment to the path
![2019-04-20_14-04-18](https://user-images.githubusercontent.com/11589750/56456482-be819e80-6375-11e9-9a44-f3df6ba64454.jpg)

**Comment by dbaeumer** [comment]
@maxim-ge can you please provide me with a GitHub repository I can clone that repros this. VS Code in general has no problems with `-` in names. So I suspsect this is either a problem with an extension or a combination of VS Code and an extension. To understand what is going on I would need a case to reproduce.

**Comment by pietie** [comment]
I've also hit this issue on multiple projects (C#). My paths also tend to have dashes in them, e.g. "00 - A\01 - B\..." but in my case VS Code tries to access the file from the drive root (e.g. C:/myfile.cs)

I tried to repro this on a new project and could not get it to fail in exactly the same way. It would fail to find a file  from the Problems tab if I rename a folder in the project from within VS Code and then rebuilt. To fix this I had to restart VS Code so this seems like a different bug.

But then I noticed that my build tasks in tasks.json are different from the new test app. When I changed the task **type** from _shell_ to _process_ on my main projects the issue went away and VS Code is now able to find and open files listed with problems.

**Comment by dbaeumer** [comment]
> I tried to repro this on a new project and could not get it to fail in exactly the same way. It would fail to find a file from the Problems tab if I rename a folder in the project from within VS Code and then rebuilt. To fix this I had to restart VS Code so this seems like a different bug.

Did these problems come from a task or from the C# extension? If the second can you report the issue there please.

**Comment by pietie** [comment]
> Did these problems come from a task or from the C# extension? If the second can you report the issue there please.

In terms of the rename problem I see the issue listed on omnisharp's repo so I guess it's a problem with the extension.

As for the original problem all I know is the task type fixed it. How would I know it it's the extension? I don't see anything in the Omnisharp Log.

**Comment by dbaeumer** [comment]
Usually a problem has an indication who generated it (e.g. ts, c#, ...)

**Comment by xhliu** [comment]
I'm having the same issue with haskell. I have to restart every time. It's really annoying. Does anyone have a solution yet? This does not work for me: https://stackoverflow.com/questions/39306348/visual-studio-code-unable-to-open-myfile-h-file-not-found
VS Code Version 1.34.0

**Comment by dbaeumer** [comment]
@xhliu does that file actually exist in our workspace?

**Comment by xhliu** [comment]
Yes

**Comment by XinChen99** [comment]
i have the same problem when editing the latex file as well. Since i had changed the folder name before and it seems that the vscode has a memory that it only remembers the old folder name, that's why i keep getting the same problem when i click the warning links, because it links to the older folder name.

**Comment by dbaeumer** [comment]
@xhliu and @XinChen99 how does the error look like in the problems view. Do you use any additional extensions that produce these errors?

**Comment by XinChen99** [comment]
> @xhliu and @XinChen99 how does the error look like in the problems view. Do you use any additional extensions that produce these errors?
![image](https://user-images.githubusercontent.com/18189010/58459825-d8758480-815e-11e9-8ee2-fc1729ef3708.png)
it looks like this, actually i had change the folder name from 'MY PAPER' to 'MY PAPER&CONFERENCE' before, but the vscode seems do not find it and the error message still links to the old path.

additional extensions? i actually don't know which extension may be responsible for the error here.

**Comment by dbaeumer** [comment]
@XinChen99 can you provide me with a GitHub repository I can clone that demos this. And which LaTeX extension are you using?

**Comment by ColdEvul** [comment]
Having a similar problem out of nowhere today version 1.36.0:
![image](https://user-images.githubusercontent.com/13811113/60627047-13697880-9ded-11e9-9c9a-173e3d7f9b37.png)

**Comment by dbaeumer** [comment]
@ColdEvul is the path name containing `%2e` or a `.`

**Comment by ColdEvul** [comment]
yes

**Comment by ColdEvul** [comment]
I am unsure if I can or are allowed to rename the folder to use `. ` instead due to engine limitations of the game using the directory (Arma3).

The issue suddenly appeard yesterday after a commit.

**Comment by ColdEvul** [comment]
Manage to reproduce it on my Ubuntu as well lifting my issue out to another issue. `%2e` causes the path to be `.` instead of the selected name.

**Comment by dbaeumer** [comment]
@jrieken please see comments above. Another case of the URI encoding / decoding problem.

**Comment by jrieken** [comment]
This bug is one year old and unrelated, the comments above are more or less the same as in #76612 and all from the same author...

**Comment by amerzafar** [comment]
Yes, this issue is very old and having this on vs code very frequently. I am wondering when will this be resolved.

**Comment by alexanderyhapcv** [comment]
having this issue as well.

![image](https://user-images.githubusercontent.com/20650028/60733735-c0daaa00-9f1b-11e9-8f28-5cf8d3b0235b.png)

**Comment by dbaeumer** [comment]
@jrieken I was referring to the comment here https://github.com/microsoft/vscode/issues/50426#issuecomment-508432817 where a user as `%2e` in its file name. His problem has nothing to do with the original issue describe here and is IMO related to the URI encoding / decoding problem.

**Comment by jdegoes** [comment]
I have this problem as well.

Here is what I have observed:

1. `Cmd + P`
2. Select file
3. Get error message
4. `Cmd + P` again
5. Select file
6. No error message!

This made me think it was an indexing problem, i.e. all the "recently opened" files were indexed somehow; and the index was corrupted; and when I failed to open a file, it cleared it from the index; then opening the file again allowed a new index to be created.

I didn't find any command to "refresh index" to test that hypothesis, however, and I didn't dig into the code to find out if `vscode` even uses an index (although given how fast it is, I'd expect so).

I don't think I have any strange symbols in the code. However, like the original poster of this issue, I am using Scala, and probably have the same set of plugs as he does (Scala Syntax from scala-lang, Metals).

**Comment by dbaeumer** [comment]
@jdegoes Scala error message are generated by an extension providing the Scala support. Out of the box VS Code is not generating any of those. Could you please file an issue against the Scala extension you are using with steps on how to reproduce. Best is always a GitHub repository to clone.

**Comment by jdegoes** [comment]
@dbaeumer To clarify, the error does not appear to be related to Scala. Rather, the error occurs on opening _any_ file from the project, and the exact error message is:

`Unable to open …: File not found`

where `...` is the full path to the file. It occurs on `Cmd + P` only, not on double-clicking the same file on the window pane; furthermore, the second attempt to open the file is always successful, within a single session.

**Comment by dbaeumer** [comment]
@jdegoes are you saying it happens when opening it from Quick Pick. If so can you please open a separate issues with steps how to reproduce. This would be unrelated to the problems view.

**Comment by alexr00** [comment]
If anyone experiencing this issue has a repo I can clone or detailed repro steps (that don't include installing an extension) I'm happy to take a look! Otherwise, this issue will be closed soon.

If you need a problem matcher from an extension for your repro the problem is likely with that problem matcher, so try opening an issue with the extension that the problem matcher comes from.

If the maintainers that extension comes back and says it isn't their problem then it could be a problem with VS Code's problem matcher logic, in which case I'd like to look at it.

Finally, I strongly recommend opening a _new issue_ if you see something that looks like this issue. This issue has a lot of potentially unrelated problems in it and isn't a good place to get things done.

**Comment by stevie1706** [comment]
`  {
            "label": "build-nameofapp",
            "command": "dotnet",
            "type": "process",
            "args": [
                "build",
                "${workspaceFolder}/src/Services/Service.API.csproj",
                "/property:GenerateFullPaths=true"
            ],
            "problemMatcher": "$msCompile"
        }`

Adding the following line to args  "/property:GenerateFullPaths=true"  fixed it for me.

**Comment by jamescornett** [comment]
@stevie1706's fix worked for me.  Was having issues with c# dotnet compiler and clicking the link in the Problems list to a error reference. Was getting file not found errors.  Adding the "/property:GenerateFullPaths=true" to the args array fixed the issue. Is this added there now in new projects?
